src/gateway/energy/energy_bus_controller.py

```diff
@@ -421,18 +423,35 @@ def _distribute_addresses(context):
 
 
     @staticmethod
-    def _find_address_for_serial(serial_number):
-        # type: (str) -> Tuple[int, bool]
+    def _find_address_for_serial(serial_number, old_address):
+        # type: (str, int) -> Tuple[int, bool]
         with Database.get_session() as db:
             module = db.query(Module).where(Module.source == ModuleDTO.Source.GATEWAY,
                                             Module.hardware_type == HardwareType.PHYSICAL,
                                             Module.serial_number == serial_number).first()
             if module is None:
-                address = EnergyBusState.get_next_unused_address()
+                # if no module has been found with serial number, check for module with address as a fallback
+                query = db.query(Module).where(Module.source == ModuleDTO.Source.GATEWAY,
```

Comment:
for simple field queries `.filter_by` can be used

```suggestion
                query = db.query(Module).filter_by(source=ModuleDTO.Source.GATEWAY,
                                                   hardware_type=HardwareType.PHYSICAL,
                                                   ...
```